crates/sui-tool/src/lib.rs

```diff
@@ -618,6 +620,103 @@ pub async fn restore_from_db_checkpoint(
     Ok(())
 }
 
+pub async fn restore_from_formal_snapshot(
+    snapshot_object_store_config: ObjectStoreConfig,
+    archive_object_store_config: ObjectStoreConfig,
+    epoch: EpochId,
+    parent_db_path: PathBuf,
+    local_store_dir: PathBuf,
+    genesis: Genesis,
+    concurrency: Option<NonZeroUsize>,
+    disable_verification: Option<bool>,
+) -> anyhow::Result<()> {
+    // first ensure that target db is empty
+    if !AuthorityPerpetualTables::open(&parent_db_path.join("store").join("perpetual"), None)
+        .database_is_empty()
+        .expect("Database read should not fail at init.")
+    {
+        panic!("Database is not empty. Will not perform snapshot restoration");
+    }
+
+    let genesis_committee = genesis.committee()?;
+
+    let perpetual_staging = Arc::new(AuthorityPerpetualTables::open(
+        &parent_db_path.join("store_staging").join("perpetual"),
+        None,
+    ));
+
+    let committee_store = Arc::new(CommitteeStore::new(
+        parent_db_path.join("epochs"),
+        &genesis_committee,
+        None,
+    ));
+
+    let store = AuthorityStore::open(
+        perpetual_staging.clone(),
+        &genesis,
+        &committee_store,
+        usize::MAX,
+        false,
+        &Registry::default(),
+    )
+    .await?;
+
+    let checkpoint_store = Arc::new(CheckpointStore::open_tables_read_write(
+        parent_db_path.join("checkpoints"),
```

Comment:
After transaction pruning, even this is not a big db and we could just copy it under `/opt/sui/db/authority_db/staging`